src/main/java/org/openlmis/buq/validate/BottomUpQuantificationValidator.java

```diff
@@ -36,6 +37,7 @@
 import org.springframework.stereotype.Component;
 import org.springframework.validation.Errors;
 
+@SuppressWarnings("PMD.TooManyMethods")
```

Comment:
Hmm, 5 public methods, PMD seems to be too restrictive?

I'm not sure how much time you have, so this is **not a request to change**, but some idea to have it better in future is split it into:
* BottomUpQuantificationCanSubmitValidator
* BottomUpQuantificationCanAuthorizeValidator
* ...
* BottomUpQuantificationCanChangeStatusValidator
Then, you call BottomUpQuantificationCanSubmit/CanAuthorize/...Validator, it checks the status or w/e else it calls BottomUpQuantificationCanChangeStatusValidator. Can be done with beans, but simple classes should be enough too.

Reply:
I agree. We may divide the validator by functionality in the future.